core/src/main/scala/kafka/utils/KafkaScheduler.scala

```diff
@@ -125,6 +127,12 @@ class KafkaScheduler(val threads: Int,
     }
   }
 
+  def taskRunning(task: ScheduledFuture[_]) : Boolean = {
```

Comment:
Maybe we should make this `private[utils]` and mention in the comment that it's package private for testing. Also, do we actually need the `synchronnized`?

Reply:
I had it there because isStarted was also synchronized, but I removed the synchronized